LandSandBoat/server
# [core] Intialize uninitialized members, don't access out of bounds memory, add constructors, small memcpy fixes, fix various warnings

**_I affirm:_**

## What does this pull request do?

Initialize previously uninitialized members, memcpy fixes, fix various warnings, add tons of constructors to prevent uninitialized members.

## Steps to test these changes

Load xi_map with valgrind and get less "jump based on uninitialized member" notifications than before. ~~This is not a "functional" PR, but intended to help other debug tools such as valgrind report more "real" problems.~~

Test a bunch of stuff to make sure it still works:
- [x] Lobby
- [x] Logging in
- [x] Unity Rankings
- [x] Fishing ~~(including size/weight of catch)~~ (most fish are zeroed)
- [x] Trust Gambits/Trusts
- [x] Pathfinding (navmesh)
- [x] Battlefields
- [x] Looks on mob/npc/dynamic entity
- [x] Instances
- [x] Linkshell encode/decode
- [x] Item Signature encode/decode
- [x] Guild shop
- [x] Status Effects
- [x] Merit menu
- [x] Map regions
- [x] ROE
- [x] Treasure pool/lotting
- [x] Pets
- [x] Dynamic Entity in town + pathing
